urlc8dbc7ce6e_dlubarov_guava_tgz-pJ8-MainJ8/src/d/BytecodeMethodDef.java

```diff
@@ -0,0 +1,467 @@
+package d;
+
+import static d.Opcodes.*;
+
+import util.StringUtils;
+
+import common.*;
+import d.nat.*;
+import d.ty.ConcreteType;
+import d.ty.desc.TypeDesc;
+import d.ty.nf.NonFinalType;
+
+public class BytecodeMethodDef extends ConcreteMethodDef {
+    public final String[] stringTable;
+    public final BaseObject[] zeptoStringTable;
+
+    private final int numLocals; // includes "this"
+    private final int[] code;
+
+    public BytecodeMethodDef(RawMethod desc,
+            RawType[] typeDescTable,
+            TypeDesc[] fullTypeDescTable,
+            RawMethod[] methodDescTable,
+            String[] stringTable,
+            int numLocals, int[] bytecode) {
+        super(desc, typeDescTable, fullTypeDescTable, methodDescTable);
+
+        for (int i = 0; i < stringTable.length; ++i)
+            stringTable[i] = stringTable[i].intern();
+        this.stringTable = stringTable;
+        zeptoStringTable = new BaseObject[stringTable.length];
+
+        this.numLocals = numLocals;
+        this.code = bytecode;
+    }
+
+    @Override
+    public void link() {
+        super.link();
+        for (int i = 0; i < zeptoStringTable.length; ++i)
+            zeptoStringTable[i] = VMUtils.makeString(stringTable[i]);
+    }
+
+    @Override
+    public void invoke(BaseObject[] stack, int bp, ConcreteType[] genericArgs) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 251 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.